### ISSUE TITLE
Bug: Indent callback function

### ISSUE THREAD

**Issue by ChristianMurphy** [eslint/eslint#4165]
**Eslint Version**
1.7.0

**Eslint configuration**
``` yaml
rules:
    indent:
        - 2
        - 4
        - SwitchCase: 1
```

**Source Code**
``` js
function example (callback) {
    callback();
}

example(
    function () {
        console.log('example');
    }
);
```

**Errors**
```
7:9  error  Expected indentation of 4 space characters but found 8  indent
8:5  error  Expected indentation of 0 space characters but found 4  indent
```

**Expected**
No Errors

**Note**
Eslint 1.6.0 does not print errors for this code.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ChristianMurphy** [comment]
likely same issue as #4164

**Comment by mjrussell** [comment]
:+1: Same thing happens when you use es6 lambdas and babel.
